cmd/vic-machine/common/cnetworks.go

```diff
@@ -26,21 +26,50 @@ import (
 	"github.com/vmware/vic/pkg/ip"
 )
 
+type TrustLevel int
+
+const (
+	Published TrustLevel = iota
+	Open
+	Closed
+	Outbound
+	Peers
+)
+
+func (t TrustLevel) String() string {
+	switch t {
+	case Open:
+		return "open"
+	case Closed:
+		return "closed"
+	case Published:
+		return "published"
+	case Outbound:
+		return "outbound"
+	case Peers:
+		return "peers"
+	}
+	log.Warningf("Unknown trust level %d. Defaulting to Published.", int(t))
+	return "published"
+}
+
 // CNetworks holds user input from container network flags
 type CNetworks struct {
 	ContainerNetworks         cli.StringSlice `arg:"container-network"`
 	ContainerNetworksGateway  cli.StringSlice `arg:"container-network-gateway"`
 	ContainerNetworksIPRanges cli.StringSlice `arg:"container-network-ip-range"`
 	ContainerNetworksDNS      cli.StringSlice `arg:"container-network-dns"`
+	ContainerNetworksFirewall cli.StringSlice `arg:"container-network-firewall"`
 	IsSet                     bool
 }
 
 // ContainerNetworks holds container network data after processing
 type ContainerNetworks struct {
-	MappedNetworks         map[string]string     `cmd:"parent" label:"key-value"`
-	MappedNetworksGateways map[string]net.IPNet  `cmd:"gateway" label:"key-value"`
-	MappedNetworksIPRanges map[string][]ip.Range `cmd:"ip-range" label:"key-value"`
-	MappedNetworksDNS      map[string][]net.IP   `cmd:"dns" label:"key-value"`
+	MappedNetworks          map[string]string     `cmd:"parent" label:"key-value"`
+	MappedNetworksGateways  map[string]net.IPNet  `cmd:"gateway" label:"key-value"`
+	MappedNetworksIPRanges  map[string][]ip.Range `cmd:"ip-range" label:"key-value"`
+	MappedNetworksDNS       map[string][]net.IP   `cmd:"dns" label:"key-value"`
+	MappedNetworksFirewalls map[string]TrustLevel `cmd:"firewall" label:"key-value"`
 }
 
 func (c *ContainerNetworks) IsSet() bool {
```

Comment:
this flag is to test if there is any container network options set. Should be changed to include container network firewall configuration as well